src/sql/CreateStatement.h

```diff
@@ -33,17 +33,42 @@ struct ColumnDefinition : TableElement {
 
   ~ColumnDefinition() override;
 
-  void setNullableExplicit() {
-    nullable = false;
+  bool trySetNullableExplicit() {
+    bool explicit_nullable = false;
+    bool explicit_not_nullable = false;
+    std::vector<unsigned long> constraints_to_remove;
 
     for (unsigned long constraint_index = 0; constraint_index < column_constraints->size(); constraint_index++) {
-      if (column_constraints->at(constraint_index) == ConstraintType::Null) {
-        nullable = true;
-        column_constraints->erase(column_constraints->cbegin() + constraint_index);
-      } else if (column_constraints->at(constraint_index) == ConstraintType::NotNull) {
-        column_constraints->erase(column_constraints->cbegin() + constraint_index);
+      const auto column_constraint = column_constraints->at(constraint_index);
+      switch (column_constraint) {
+        case ConstraintType::Null:
+          if (explicit_not_nullable) {
+            return false;
+          }
+          explicit_nullable = true;
+          constraints_to_remove.emplace_back(constraint_index);
+          break;
+        case ConstraintType::NotNull:
+          constraints_to_remove.emplace_back(constraint_index);
+          [[fallthrough]];
+        case ConstraintType::PrimaryKey:
+          if (explicit_nullable) {
+            return false;
+          }
+          explicit_not_nullable = true;
+          nullable = false;
+          break;
+        default:
+          break;
       }
     }
+
+    for (auto constraint_index = constraints_to_remove.rbegin(); constraint_index != constraints_to_remove.rend();
```

Comment:
We may want to keep this information and delete it later. See my overall comment.

Reply:
I do not delete the information anymore. This way, the DBMS can check if a column is explicitly set to be nullable for a PRIMARY KEY table constraint.